# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/BaselineDebugModeOSR.h
# Commit: 49a2da59aa3e
# Full Hash: 49a2da59aa3e33b3719c7953a98fac6c24b8a514
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-07-21 21:59:35
# Regressor Bug: 1566330
# File Overlap Count: 2
# Description:
#   Bug 1566330 - Let BaselineDebugModeOSR resume in the interpreter, remove BaselineDebugModeOSRInfo. r=iain
#   
#   At this point most of the DebugModeOSR complexity came from dealing with the
#   On -> Off case because debugger callVMs are not present in the recompiled script.
#   We also had to worry about loading unsynced stack values in R0/R1 in the
# ==============================================================================

diff -r ad5e0980c273 -r 49a2da59aa3e js/src/jit/BaselineDebugModeOSR.h
--- a/js/src/jit/BaselineDebugModeOSR.h	Sun Jul 21 09:27:47 2019 +0000
+++ b/js/src/jit/BaselineDebugModeOSR.h	Fri Jul 19 09:01:45 2019 +0000
@@ -47,32 +47,6 @@
                                    uint8_t* newAddr);
 };
 
-//
-// Auxiliary info to help the DebugModeOSRHandler fix up state.
-//
-struct BaselineDebugModeOSRInfo {
-  uint8_t* resumeAddr;
-  jsbytecode* pc;
-  PCMappingSlotInfo slotInfo;
-  RetAddrEntry::Kind frameKind;
-
-  // Filled in by SyncBaselineDebugModeOSRInfo.
-  uintptr_t stackAdjust;
-  Value valueR0;
-  Value valueR1;
-
-  BaselineDebugModeOSRInfo(jsbytecode* pc, RetAddrEntry::Kind kind)
-      : resumeAddr(nullptr),
-        pc(pc),
-        slotInfo(0),
-        frameKind(kind),
-        stackAdjust(0),
-        valueR0(UndefinedValue()),
-        valueR1(UndefinedValue()) {}
-
-  void popValueInto(PCMappingSlotInfo::SlotLocation loc, Value* vp);
-};
-
 MOZ_MUST_USE bool RecompileOnStackBaselineScriptsForDebugMode(
     JSContext* cx, const Debugger::ExecutionObservableSet& obs,
     Debugger::IsObserving observing);